inc/classes/subscriber/Cache/class-expired-cache-purge-subscriber.php

```diff
@@ -88,12 +88,31 @@ public function clean_expired_cache_scheduled_event( $old_value, $value ) {
 		if ( empty( $value['purge_cron_unit'] ) ) {
 			return;
 		}
-		// If Lifespan unit is not changed to minutes do not reschedule (for hours & days event should be scheduled at 1 hour).
-		if ( 'MINUTE_IN_SECONDS' !== $value['purge_cron_unit'] ) {
+
+		$unit_list = [ 'HOUR_IN_SECONDS', 'DAY_IN_SECONDS' ];
+		// Bail out if the cron unit is changed from hours to days.
+		if ( $old_value['purge_cron_unit'] === $value['purge_cron_unit'] &&
+				in_array( $old_value['purge_cron_unit'], $unit_list, true ) &&
+				in_array( $value['purge_cron_unit'], $unit_list, true ) ) {
```

Comment:
One of those `in_array()` is useless.

Reply:
Why do you think is useless?

Reply:
If A = B and A is in C, then B is in C.

Reply:
Indeed, this condition should be totally removed because the next ones prevent this.
I will make a PR with origin:develop branch

Reply:
Do like you want, it doesn't hurt, it can stay here until there is something else to do.